src/shared_memory.rs

```diff
@@ -1,18 +1,31 @@
-pub trait ShareableMemory {
-    fn driver_number(&self) -> usize;
+use core::ptr;
+use core::slice;
+use syscalls;
 
-    fn allow_number(&self) -> usize;
-
-    fn to_bytes(&mut self) -> &mut [u8];
+pub struct SharedMemory<'a> {
+    pub driver_number: usize,
+    pub allow_number: usize,
+    pub buffer_to_share: &'a mut [u8],
 }
 
-pub struct SharedMemory<SM: ShareableMemory> {
-    #[allow(dead_code)] // Used in drop
-    pub(crate) shareable_memory: SM,
+impl<'a> SharedMemory<'a> {
+    pub fn read_bytes(&self, destination: &mut [u8]) {
+        destination.clone_from_slice(self.buffer_to_share);
+    }
+
+    pub fn write_bytes(&mut self, source: &[u8]) {
+        self.buffer_to_share.clone_from_slice(source);
+    }
 }
```

Comment:
Those function panic if the lengths of the buffers differ. I don't know if this is what a user would expect.